tests/functional/test_install_config.py

```diff
@@ -352,12 +353,50 @@ def test_do_not_prompt_for_authentication(
     assert "ERROR: HTTP error 401" in result.stderr
 
 
-@pytest.mark.parametrize("auth_needed", (True, False))
+@pytest.fixture(params=(True, False), ids=("auth_needed", "auth_not_needed"))
+def auth_needed(request: pytest.FixtureRequest) -> bool:
+    return request.param  # type: ignore[attr-defined]
```

Comment:
It was fixed in https://github.com/pytest-dev/pytest/pull/10133, why pytest is pinned in pre-commit config and not test requirements?

Reply:
No idea. As far as I can tell it is just for the type annotations. Want me to unpin it or pin it to 7.1.3? Or create a new issue/PR and then rebase after that is resolved? Or nothing/something else?

Reply:
I prefer to unpin pytest in .pre-commit-config.yaml here

Reply:
Done